client/service/src/client/wasm_override.rs

```diff
@@ -35,29 +35,43 @@
 //! A custom WASM blob will override on-chain WASM if the spec version matches. If it is
 //! required to overrides multiple runtimes, multiple WASM blobs matching each of the spec versions
 //! needed must be provided in the given directory.
+
 use sc_executor::RuntimeVersionOf;
 use sp_blockchain::Result;
-use sp_core::traits::{FetchRuntimeCode, RuntimeCode};
+use sp_core::traits::{FetchRuntimeCode, RuntimeCode, WrappedRuntimeCode};
 use sp_state_machine::BasicExternalities;
 use sp_version::RuntimeVersion;
 use std::{
 	collections::{hash_map::DefaultHasher, HashMap},
 	fs,
 	hash::Hasher as _,
 	path::{Path, PathBuf},
+	time::{Duration, Instant},
 };
 
-#[derive(Clone, Debug, PartialEq)]
+/// The interval in that we will print a warning when a wasm blob `spec_name`
+/// doesn't match with the on-chain `spec_name`.
+const WARN_INTERVAL: Duration = Duration::from_secs(30);
```

Comment:
wouldn't it be much easier if we just abort the command if this didn't match?

Reply:
it would spare you most of the Arc/Mutex layers as well.

Reply:
> wouldn't it be much easier if we just abort the command if this didn't match?

Which command? 

I mean the best would be to do that at startup, but there we don't know the spec name and that is also legal to change.

Reply:
Printing just one warning, will probably not bring that much, as it can be overseen relative easily.